### PR TITLE
add function to save model to PB

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: elasticdl/pkg/ps/model.go [hunk 1]
```diff
@@ -71,3 +72,25 @@ func (model *Model) InitFromModelPB(pb *proto.Model) error {
 	}
 	return nil
 }
+
+// SaveToModelPB saves in memory model to PB
```
Comment (skydoorkai) [inline elasticdl/pkg/ps/model.go]: `saves in memory model` -> `saves in-memory model`

Comment (QiJune) [reply elasticdl/pkg/ps/model.go]: Done

File: elasticdl/pkg/common/embedding_table.go [hunk 2]
```diff
@@ -50,3 +50,15 @@ func (e *EmbeddingTable) SetEmbeddingVectors(idxslice *IndexedSlices) error {
...
+// ToIndexedSlices transforms embedding table format to indexed slices format
+func (e *EmbeddingTable) ToIndexedSlices() *IndexedSlices {
+	ids := make([]int64, 0, len(e.EmbeddingVectors))
+	for k := range e.EmbeddingVectors {
+		ids = append(ids, k)
+	}
+	return &IndexedSlices{
+		ConcatTensors: e.GetEmbeddingVectors(ids),
+		Ids:           ids,
+	}
```
Comment (lwpyr) [inline elasticdl/pkg/common/embedding_table.go]: How about  `NewIndexedSlices(e.ConcatTensors, ids)`

File: elasticdl/pkg/ps/model_test.go [hunk 3]
```diff
@@ -75,3 +75,34 @@ func TestModelInitFrom(t *testing.T) {
...
+
+	d1 := []int64{3, 2}
+	v1 := []float32{1.0, 2.0, 3.0, 4.0, 5.0, 6.0}
+	t1 := common.NewTensor(v1, d1)
+
+	i1 := []int64{1, 3, 5}
+	var is = common.IndexedSlices{
+		ConcatTensors: t1,
+		Ids:           i1,
+	}
```
Comment (lwpyr) [inline elasticdl/pkg/ps/model_test.go]: How about using `NewIndexedSlices` function

File: elasticdl/pkg/common/embedding_table.go (hunk 2 above)
Comment (QiJune) [reply elasticdl/pkg/common/embedding_table.go]: Done

File: elasticdl/pkg/ps/model_test.go (hunk 3 above)
Comment (QiJune) [reply elasticdl/pkg/ps/model_test.go]: Done

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**